substrate/primitives/crypto/ec-utils/src/utils.rs

```diff
@@ -25,17 +25,18 @@ use ark_ec::{
 	twisted_edwards::TECurveConfig,
 	CurveConfig, VariableBaseMSM,
 };
-use ark_scale::hazmat::ArkScaleProjective;
-use ark_std::vec::Vec;
-use codec::{Decode, Encode};
+use ark_scale::{
+	hazmat::ArkScaleProjective,
+	scale::{Decode, Encode},
+};
+use sp_std::vec::Vec;
 
-const HOST_CALL: ark_scale::Usage = ark_scale::HOST_CALL;
-type ArkScale<T> = ark_scale::ArkScale<T, HOST_CALL>;
+// Scale codec type which is expected to be used by the host functions.
+//
+// Encoding is set to `HOST_CALL` which is a shortcut for "not-validated" and "not-compressed".
+type ArkScale<T> = ark_scale::ArkScale<T, { ark_scale::HOST_CALL }>;
 
-pub(crate) fn multi_miller_loop_generic<Curve: Pairing>(
-	g1: Vec<u8>,
-	g2: Vec<u8>,
-) -> Result<Vec<u8>, ()> {
+pub fn multi_miller_loop<Curve: Pairing>(g1: Vec<u8>, g2: Vec<u8>) -> Result<Vec<u8>, ()> {
```

Comment:
All these functions could have stayed `pub(crate)`?

Reply:
As they don't have any usage outside of this crate, I'd answer yes.
I removed `pub(crate)` because the `utils` module itself is not public.

Reply:
Yeah fine.